### PR TITLE
refactor: breakout config

### PR DESCRIPTION
**Pull request by tylerslaton**
Breaking out the `Config` struct into its own file to reduce the amount of code in `lib.rs` to just `run()`. This will make it a nicer experience when having to edit the core application logic (`run()`) or the input it accepts (`Config`).

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
+use anyhow::{anyhow, Result};
+use clap::{App, Arg};
+
+use crate::runtime;
+
+pub const VERSION: &str = "0.4.0";
```
Comment (exdx) [inline src/config.rs:6]: nit: we can probably drop the type annotation here -- rustc will infer it for us, and it doesn't provide much value.

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+#[derive(Debug)]
+pub struct Config {
```
Comment (exdx) [inline src/config.rs:9]: I think `config::Config` isn't an ideal import path, but nothing better comes to mind at the moment. We could refactor in a follow-up.

**Review by exdx — APPROVED** [review]

File: src/config.rs (hunk 1 above)
Comment (tylerslaton) [reply src/config.rs:6]: Ah, right right.

File: src/config.rs (hunk 2 above)
Comment (tylerslaton) [reply src/config.rs:9]: I couldn't think of anything better either. If it annoys us a bunch in the future we can switch it up.

File: src/config.rs (hunk 1 above)
Comment (tylerslaton) [reply src/config.rs:6]: Screen Shot 2022-10-04 at 11 04 28 AM

Ah, just remembering. This is why I added the type annotation here. I was reading a bit about the _why_ behind this [here](https://users.rust-lang.org/t/why-is-type-declaration-necessary-in-constants/14200/3).

**Pull request merged**